src/front/glsl/builtins.rs

```diff
@@ -2099,3 +2039,71 @@ pub fn sampled_to_depth(
         }),
     };
 }
+
+/// Helper function to generate the image components for texture/image builtins
+///
+/// Calls the passed function `f` with:
+/// ```text
+/// f(ScalarKind, ImageDimension, arrayed, multi, shadow)
+/// ```
+///
+/// The generation of the shadow component can be controlled with `generate_shadow`,
+/// if it's `true` then shadow will be passed as `true` and `false`, otherwise only\
+/// `false` is passed.
+///
+/// Likewise `generated_multi` controls the generation of the multi component.
+fn texture_args_generator(
+    generate_multi: bool,
```

Comment:
boolean arguments should generally not be used
did you consider bitflags instead?